File: data/makeData.py
import csv
import random
import os

filename = '2gb_file.csv'
size_limit = 2 * 1024 * 1024 * 1024  # 2 GB
header = ['Column1', 'Column2', 'Column3', 'Column4', 'Column5']

def generate_row():
    return [
        random.randint(1, 100000),
        random.uniform(0, 100),
        random.choice(['A', 'B', 'C', 'D', 'E']),
        random.choice(['X', 'Y', 'Z']),
        random.randint(1, 1000)
    ]

with open(filename, 'w', newline='') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(header)

    while os.path.getsize(filename) < size_limit:
        writer.writerow(generate_row())

File: data/makeData.js
const fs = require("fs");
const path = require("path");

const filename = "20230328.csv";
const sizeLimit = 2 * 1024 * 1024 * 1024; // 2 GB
const header = ["Column1", "Column2", "Column3", "Column4", "Column5"];

function generateRow() {
  return [
    Math.floor(Math.random() * 100000),
    (Math.random() * 100).toFixed(2),
    ["A", "B", "C", "D", "E"][Math.floor(Math.random() * 5)],
    ["X", "Y", "Z"][Math.floor(Math.random() * 3)],
    Math.floor(Math.random() * 1000),
  ].join(",");
}

fs.writeFileSync(filename, header.join(",") + "\n");

while (fs.statSync(filename).size < sizeLimit) {
  fs.appendFileSync(filename, generateRow() + "\n");
}
